test/integration/conditions.test.ts

```diff
@@ -262,7 +267,75 @@ describe('condition context', () => {
       web3Provider
     );
     const asJson = await conditionContext.toJson();
-    expect(asJson).toBeDefined();
+    expect(asJson).toContain(USER_ADDRESS_PARAM);
+  });
+
+  describe('supports user-defined parameters', () => {
+    const fakeFunctionAbi = {
+      name: 'myFunction',
+      type: 'function',
+      inputs: [
+        {
+          name: 'account',
+          type: 'address',
+        },
+        {
+          name: 'myCustomParam',
+          type: 'uint256',
+        },
+      ],
+      outputs: [
+        {
+          name: 'someValue',
+          type: 'uint256',
+        },
+      ],
+    };
+    const evmCondition = new Conditions.EvmCondition({
+      chain: 5,
+      functionAbi: fakeFunctionAbi,
+      method: 'balanceOf',
+      contractAddress: '0x0000000000000000000000000000000000000000',
+      parameters: [USER_ADDRESS_PARAM, ':customParam'],
+      returnValueTest: {
+        index: 0,
+        comparator: '==',
+        value: USER_ADDRESS_PARAM,
```

Comment:
Does a user-defined variable work here (with the returnValuetest) instead of `USER_ADDRESS_PARAM`? eg. if you instead had:

```suggestion
        value: ':customParam',
```

I saw the code for processing when it is in params, but I haven't seen it for return value test objects - I may have missed it somewhere (my typescript is not great 😅 ).

Reply:
Yes, it does work here. IMO this is a desirable behavior - being able to provide custom parameters for predefined variables. 

I'm adding a test to document that behavior.